tests/core/test_linalg.py

```diff
@@ -125,6 +125,19 @@ def test_inv(app_inst: ArrayApplication):
         assert np.allclose(np.linalg.cholesky(mat.get()), L, rtol=1e-4, atol=1e-4)
 
 
+def test_inv_uppertri(app_inst: ArrayApplication):
```

Comment:
Add assertions for tests here so that the input block shape matches the output block shape.

Also, can we include a test on less trivial input? I think you can obtain an invertible upper triangular matrix with `_, R = np.qr(np.random.rand(20, 20))`.